packages/flutter/test/material/dropdown_form_field_test.dart

```diff
@@ -195,6 +195,162 @@ void main() {
+                value: value,
+                child: Text(value),
+              );
+            }).toList(),
+          ),
+        ),
+      )
+    );
+
+    // [FormField] value is properly initialized.
+    expect(stateKey.currentState.value, equals('one'));
+
+    // Tap [DropdownButton] to open dropdown menu.
+    await tester.tap(find.text('one'));
+    await tester.pumpAndSettle();
+
+    // Tap [DropdownMenuItem] for an item other than the initial item.
+    await tester.tap(find.text('three').last);
+    await tester.pumpAndSettle();
+
+    // [FormField] should update to selected item value.
+    expect(stateKey.currentState.value, equals('three'));
+  });
+
+  testWidgets("DropdownButtonFormField's InputDecorator isEmpty behavior is correct", (WidgetTester tester) async {
+    final GlobalKey<FormFieldState<String>> stateKey = GlobalKey<FormFieldState<String>>();
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: DropdownButtonFormField<String>(
+            key: stateKey,
+            value: null, // setting value to null sets isEmpty to true
+            decoration: const InputDecoration(labelText: 'label', alignLabelWithHint: true),
+            onChanged: (String newValue) {},
+            items: menuItems.map((String value) {
+              return DropdownMenuItem<String>(
+                value: value,
+                child: Text(value),
+              );
+            }).toList(),
+          ),
+        ),
+      )
+    );
+
+    // isEmpty: true causes the label to be aligned with the input text if alignLabelWithHint is set to true
+    await tester.pumpAndSettle();
+    expect(tester.getSize(find.byType(InputDecorator)), const Size(800.0, 600.0));
+    expect(tester.getTopLeft(find.text('label')).dy, 32.0);
+    expect(tester.getBottomLeft(find.text('label')).dy, 48.0);
+    expect(
+      tester.getTopLeft(find.byType(InputDecorator)).dy,
+      tester.getTopLeft(find.text('label')).dy - 32.0
+    ); // Label is lower by 32 pixels than the InputDecorator, in the center of the TextField
+    expect(
+      tester.getTopLeft(find.byType(InputDecorator)).dx,
+      tester.getTopLeft(find.text('label')).dx
+    ); // Label is aligned in x direction with the InputDecorator
+
+    // [FormField] value is initially null.
+    expect(stateKey.currentState.value, equals(null));
+    // [InputDecorator] isEmpty property is true.
+    expect((tester.widget(find.byType(InputDecorator)) as InputDecorator).isEmpty, equals(true));
+
+    // Tap [DropdownButton] to open dropdown menu.
+    await tester.tap(find.text('label'));
+    await tester.pumpAndSettle();
+
+    // Tap outside the [DropdownButton] to close dropdown menu without selecting an option.
+    await tester.tapAt(Offset.zero);
+    await tester.pumpAndSettle();
+
+    expect(tester.getTopLeft(find.text('label')).dy, 32.0);
+    expect(tester.getBottomLeft(find.text('label')).dy, 48.0);
+
```

Comment:
You might want to add the positioning checks here as well to keep your tests consistent throughout. By that, I mean the tests that look like:
```
    expect(
      tester.getTopLeft(find.text('label')).dy,
      tester.getTopLeft(find.byType(InputDecorator)).dy +/- y,
    ); // Label is lower by 12 pixels than the InputDecorator, above the TextField
    expect(
      tester.getTopLeft(find.text('label')).dx,
      tester.getTopLeft(find.byType(InputDecorator)).dx +/- x,
    ); // Label is aligned in x direction with the InputDecorator
```